sdk/identity/identity/src/util/tracing.ts

```diff
@@ -1,54 +1,53 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { GetTokenOptions } from "@azure/core-http";
-import { getTracer, OperationTracingOptions } from "@azure/core-tracing";
-import { Span, SpanKind, SpanOptions as OTSpanOptions } from "@opentelemetry/api";
+import { OperationOptions } from "@azure/core-http";
+import { createSpanFunction, SpanStatusCode, Span } from "@azure/core-tracing";
 
 /**
  * Creates a span using the global tracer.
- * @param name - The name of the operation being performed.
- * @param options - The options for the underlying http request.
+ * @internal
  */
-export function createSpan(
-  operationName: string,
-  options: GetTokenOptions = {}
-): { span: Span; options: GetTokenOptions } {
-  const tracer = getTracer();
-
-  const tracingOptions: OperationTracingOptions = {
-    spanOptions: {},
-    ...options.tracingOptions
-  };
+export const createSpan = createSpanFunction({
+  packagePrefix: "Azure.Identity",
+  namespace: "Microsoft.AAD"
+});
 
-  const spanOptions: OTSpanOptions = {
-    ...tracingOptions.spanOptions,
-    kind: SpanKind.INTERNAL
-  };
+/**
+ * From: https://github.com/Azure/azure-sdk-for-js/blob/46139daa3317a0d12e8b55b02b9d9cdf1b2e762a/sdk/appconfiguration/app-configuration/src/internal/tracingHelpers.ts
+ * Traces an operation and properly handles reporting start, end and errors for a given span
+ *
+ * @param operationName - Name of a method in the TClient type
+ * @param options - An options class, typically derived from \@azure/core-http/RequestOptionsBase
+ * @param fn - The function to call with an options class that properly propagates the span context
+ *
+ * @internal
+ */
+export async function trace<ReturnT>(
```

Comment:
I basically replicated this file from master - we don't use 'trace' in this PR so it's just a vestigial artifact....

Reply:
Can we remove it then?

Reply:
Gone!